amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamARNUtilTest.java

```diff
@@ -0,0 +1,70 @@
+package software.amazon.kinesis.common;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.regions.Region;
+
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ StreamARNUtil.class })
+public class StreamARNUtilTest {
+    private static final String STS_RESPONSE_ARN_FORMAT = "arn:aws:sts::%s:assumed-role/Admin/alias";
+    private static final String KINESIS_STREAM_ARN_FORMAT = "arn:aws:kinesis:us-east-1:%s:stream/%s";
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        PowerMockito.spy(StreamARNUtil.class);
```

Comment:
I recommend intercepting the `StsClient` constructor so you can return a mock `StsClient` object. This will "simplify" the interactions so your tests can validate expectations on a mock instead of a `private static` method. (see additional comments in this class for why that's important)

Reply:
Unfortunately, it seems that intercepting StsClient using PowerMock.whenNew is not working for some reason. One possibility is to wrap the StsClient and its getCallerIdentity method in a separate class